python/lsst/pipe/base/_task_metadata.py

```diff
@@ -392,26 +395,30 @@ def _getKeys(key):
             raise KeyError(f"Invalid key '{key}': only string keys are allowed") from None
         return keys
 
-    def keys(self):
+    def keys(self) -> Tuple[str, ...]:
```

Comment:
Probably, but `items` is actually much more problematic because it's an iterator, not a multi-pass sized iterable.

I think you can just return `KeysView(self)`, since it seems you implement a compatible `__iter__`, and returning `ValuesView(self)` and `ItemsView(self)` should work, too, if your `__getitem__` isn't too weird.

I'm not currently planning to do that on this ticket, though.